README.md

```diff
@@ -1,7 +1,331 @@
+
+The package is signed before being published to NuGet to avoid pulling a malicious package.
+We provide the `PDBs` to have a better debugging experience.
+Documentation is available for most of public members and methods.
 
 ## How to use the SDK
-Coming soon!
+First, you need to instantiate the client to interact with the platform. Here is the minimum configuration you need to provide:
+
+
+```csharp
+string apiKey = "Your api key with required rights";
+string organizationId = "You organization ID";
+ICoveoPlatformConfig config = new CoveoPlatformConfig(apiKey, organizationId);
+ICoveoPlatformClient client = new CoveoPlatformClient(config);
+```
+
+By default, the package target the US production environment (platform.cloud.coveo.com). If you want to target the US HIPAA environment (platformhipaa.cloud.coveo.com) you can do it that way:
+
+```csharp
+string apiKey = "Your api key with required rights";
+string organizationId = "You organization ID";
+ICoveoPlatformConfig config = new CoveoPlatformConfig(Constants.Endpoint.UsEast1.HIPAA_PUSH_API_URL,
+    Constants.PlatformEndpoint.UsEast1.HIPAA_PLATFORM_API_URL,
+    apiKey,
+    organizationId);
+ICoveoPlatformClient client = new CoveoPlatformClient(config);
+```
+
+## Using the SDK to push documents to a Push API source
+Methods to interact with the Push API are part of `client.DocumentManager` object.
+
+### Pushing a single document
+You should only use this method when you want to add or update a single document. Pushing several documents using this method may lead to the `429 - Too Many Requests` response from the Coveo platform and decrease the overall performance time of your application.
+
+```csharp
+string sourceId = "your Push API source ID";
+PushDocument document = new PushDocument("https://coveo.com") {
+    ClickableUri = "https://www.coveo.com",
+    ModifiedDate = DateTime.UtcNow
+};
+
+document.AddMetadata("title", "Coveo's home page.");
+
+PushDocumentHelper.SetContent(document, "this is Coveo's website home page.");
+
+client.DocumentManager.AddOrUpdateDocument(sourceId, document, null);
+```
+**Good to know:**
+* The `SetContent` method has an overload taking a Stream instead of a string.
+* The PushDocumentHelper class has also the method `SetContentFromFile` taking a file path.
+* The third argument in `AddOrUpdateDocument` is the ordering ID. If you don't provide a value, the SDK will create one using a timestamp to ensure the changes are performed in the order they were received. For more information about ordering ID, visit [Understanding the orderingId Parameter](https://docs.coveo.com/en/147/cloud-v2-developers/understanding-the-orderingid-parameter).
+* The call returns the generated ordering ID if you did not specify one. You can store it. It can be useful to delete a batch of documents.
+
+### Pushing a batch of documents
+For overall performance, it is better to push your documents in batch.
```

Comment:
I would place batch first and add "... Use the single document method when batches are not required to meet both your performance and volume requirement." Or something along those lines.